### PR TITLE
[INDY-2057] DinD for system tests

### PR DESCRIPTION
**Pull request by andkononykhin**
- improves and automates docker routine
- automates tests run inside dind container

### CODE REVIEW

File: system/docker/client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,68 @@
+FROM teracy/ubuntu:16.04-dind-latest
```
Comment (skhoroshavin) [inline system/docker/client/Dockerfile]: It is nice to have required ubuntu version and dind in one container, however AFAIK there is another quire widespread and somewhat more secure option - use official dind container + whatever container you need (in this case official ubuntu container) and call docker daemon via local network. If I'm not mistaken this is what [testinfra](https://github.com/philpep/testinfra) does.

Comment (andkononykhin) [reply system/docker/client/Dockerfile]: @skhoroshavin My understanding:
- there is no official dind image except for alpine and ubuntu 14 (if i'm not mistaken)
- testinfra doesn't implement docker client but instead uses shell to run oficial client under the hood https://testinfra.readthedocs.io/en/latest/backends.html
Could you explain your option in more details? Thanks

**Comment by andkononykhin** [comment]
Shell scrips partly duplicate what is already presented in tests setup routine (node image build and network creation) and yes it could be avoided with some better tests configuration options (version of installed artifacts etc.). But it it would be anyway necesary to build docker image for client on the host system.

Regarding unofficial image: there are options but they need some more efforts and should be discussed.

**Comment by andkononykhin** [comment]
@skhoroshavin I created tasks related to the discussion: https://jira.hyperledger.org/browse/INDY-2125 and https://jira.hyperledger.org/browse/INDY-2126. I hope the further improvements would happen in their scopes.

**Pull request merged**